android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/home/views/HomeSurveysView.kt

```diff
@@ -111,28 +126,54 @@ private fun HomeSurveysContent(uiModel: HomeSurveysUiModel) {
             ) {
-                DotsIndicator(uiModel.totalPages, uiModel.currentPageIndex)
-                Spacer(modifier = Modifier.height(AppTheme.dimensions.mediumPadding))
-                Text(
-                    text = uiModel.currentSurveyUiModel.title,
-                    color = Color.White,
-                    style = Typography.h5,
-                    maxLines = 4
+                DotsIndicator(
+                    uiModel.totalPages,
+                    uiModel.currentPageIndex,
+                    modifier = Modifier
+                        .semantics { contentDescription = HomeContentDescription.INDICATOR }
                 )
+                Spacer(modifier = Modifier.height(AppTheme.dimensions.mediumPadding))
+                AnimatedContent(
+                    targetState = uiModel.currentSurveyUiModel.title,
+                    transitionSpec = {
+                        fadeIn() with fadeOut()
+                    }
+                ) { text ->
+                    Text(
+                        text = text,
+                        color = Color.White,
+                        style = Typography.h5,
+                        maxLines = 4,
+                        modifier = Modifier
+                            .semantics { contentDescription = HomeContentDescription.SURVEY_TITLE }
+                    )
+                }
                 Spacer(modifier = Modifier.width(5.dp))
                 Row(
-                    verticalAlignment = Alignment.Bottom,
+                    horizontalArrangement = Arrangement.SpaceBetween,
+                    verticalAlignment = Alignment.CenterVertically,
                     modifier = Modifier
                         .fillMaxWidth()
                         .padding(bottom = 54.dp)
                 ) {
-                    Text(
-                        text = uiModel.currentSurveyUiModel.description,
-                        color = Color.White.copy(alpha = 0.7f),
-                        style = Typography.subtitle1,
-                        maxLines = 2,
-                        modifier = Modifier.weight(1f)
-                    )
+                    AnimatedContent(
+                        targetState = uiModel.currentSurveyUiModel.description,
+                        transitionSpec = {
+                            fadeIn() with fadeOut()
+                        }
+                    ) { text ->
+                        Text(
+                            text = text,
+                            color = Color.White.copy(alpha = 0.7f),
+                            style = Typography.subtitle1,
+                            maxLines = 2,
+                            modifier = Modifier
+                                .weight(1f)
+                                .semantics {
+                                    contentDescription = HomeContentDescription.SURVEY_DESCRIPTION
+                                }
+                        )
```

Comment:
Could you please try with the long text if it will be limited to 2 lines and end it `...`? I would suggest adding `overflow = TextOverflow.Ellipsis` 

```
Text(
    text = "Compose Previews are great to check quickly how a composable layout looks like",
    maxLines = 2,
    overflow = TextOverflow.Ellipsis
)
```

Ref: https://stackoverflow.com/a/65736376